spec/fixtures/sources/commonjs-params/index.js

```diff
@@ -0,0 +1,21 @@
+const functions = require("../../../../src/index");
+const functionsv2 = require("../../../../src/v2/index");
+const { defineString } = require("../../../../src/v2/params");
+
+defineString("foo");
```

Comment:
nit* I think our guidance will be to use capital letters when defining params. So `FOO` instead of `foo` here?

Reply:
This is a really interesting use case where even a team member got tripped up on our proposal. What should we do?

1. Throw a runtime error if the param name has invalid characters
2. Try to use the type system to require that this an uppercase string
3. Magically transform the given name into valid characters (in this case FOO)

Reply:
My vote would be for **1** (specifically runtime error that's picked up and nicely presented by the CLI) and **2**.

I feel that I've always been against magic transformation like **3** though I'm slowly turning myself around it (e.g. magically transform function name for v2)